internal/bminventory/inventory.go

```diff
@@ -732,17 +724,17 @@ func (b *bareMetalInventory) DeregisterClusterInternal(ctx context.Context, para
 
 func (b *bareMetalInventory) DownloadClusterISO(ctx context.Context, params installer.DownloadClusterISOParams) middleware.Responder {
 	log := logutil.FromContext(ctx, b.log)
-	var cluster common.Cluster
+	var infraEnv common.InfraEnv
 
-	if err := b.db.First(&cluster, "id = ?", params.ClusterID).Error; err != nil {
-		log.WithError(err).Errorf("failed to get cluster %s", params.ClusterID)
+	if err := b.db.First(&infraEnv, "id = ?", params.ClusterID).Error; err != nil {
+		log.WithError(err).Errorf("failed to get infra env %s", params.ClusterID)
 		return common.NewApiError(http.StatusNotFound, err)
```

Comment:
It can be internal error as well. 
```suggestion
		return common.GenerateErrorResponder(err)
```

Reply:
fixed